gllgo: spurious "multiple definition of `.import$1:.import$1$1$recover'" error

I am on commit a143e91 in libgo branch.

The program consists of 2 packages: "main":
https://gist.github.com/dvyukov/9f31ca735af75a18a3b1
and "b":
https://gist.github.com/dvyukov/18bc92787002840ce8f7

$ go build -compiler=gccgo main
/tmp/go-build518742582/libb.a(b.o): In function `.import$1:.import$1$1$recover':
b:(.text+0x450): multiple definition of`.import$1:.import$1$1$recover'
/tmp/go-build518742582/a/_obj/main.o:main:(.text+0x7a0): first defined here
collect2: error: ld returned 1 exit status
gllgo: error: exit status 1

--- a/ssa.go
+++ b/ssa.go
@@ -418,6 +418,7 @@ func (fr *frame) bridgeRecoverFunc(llfn llvm.Value, fti functionTypeInfo) *frame
 	ftiRecover.functionType = llvm.FunctionType(returnType, argTypes, false)
 	llfnRecover := ftiRecover.declare(fr.module.Module, llfn.Name()+"$recover")
 	addCommonFunctionAttrs(llfnRecover)
+	llfnRecover.SetLinkage(llvm.InternalLinkage)
 	args := make([]llvm.Value, len(argTypes)-1, len(argTypes))
 	for i := range args {
 		args[i] = llfn.Param(i)

